hudi-spark-datasource/hudi-spark2/src/test/java/org/apache/hudi/internal/TestHoodieDataSourceInternalWriter.java

```diff
@@ -47,12 +54,18 @@
 
   @Test
   public void testDataSourceWriter() throws Exception {
+    testDataSourceWriterInternal(Collections.EMPTY_MAP, Collections.EMPTY_MAP);
+  }
+
+  private void testDataSourceWriterInternal(Map<String, String> extraMetadata, Map<String, String> expectedExtraMetadata) throws Exception {
     // init config and table
     HoodieWriteConfig cfg = getConfigBuilder(basePath).build();
     String instantTime = "001";
     // init writer
+
     HoodieDataSourceInternalWriter dataSourceInternalWriter =
-        new HoodieDataSourceInternalWriter(instantTime, cfg, STRUCT_TYPE, sqlContext.sparkSession(), hadoopConf);
+        new HoodieDataSourceInternalWriter(instantTime, cfg, STRUCT_TYPE, sqlContext.sparkSession(), hadoopConf,
+            new DataSourceOptions(extraMetadata));
```

Comment:
Not sure why we are making options using extraMetadata. feels bit confusing

Reply:
I could not find a better way to pass user params to our custom data source classes. I mean, for those that are not part of HoodieWriteConfig, we don't pass any params as such.